rle-bitpacked.md

```diff
@@ -0,0 +1,120 @@
+# RLE-Bitpacked hybrid encoder
+
+The RLE-Bitpacked hybrid encoder is a parquet-specific encoder that combines
+two well known encoding strategies, [RLE](https://en.wikipedia.org/wiki/Run-length_encoding)
+and bitpacking. Note that "combine" here means this encoder allows both encodings
+within the same stream, and, during encoding, it can switch between them.
+
+This encoder is only used to encode integer values that may either represent definition levels,
+representation levels or ids of dictionary-encoded pages. Note that this encoder
+supports integers that can be represented in less than 8 bits.
+
+This document uses [LSB](https://en.wikipedia.org/wiki/Bit_numbering#Least_significant_bit)
+to identify bits. In this representation, a byte is represented
+by `[b7 b6 b5 b4 b3 b2 b1 b0]` where `b0` is the first bit.
+
+This document uses MUST, SHOULD, etc. according to [RFC-8174](https://tools.ietf.org/html/rfc8174).
+
+## Decoding
+
+Decoding a stream of bytes (denoted as `[a1, a2, a3, ...]`) assumes a specific `bit_width`
+indicating the number of bits necessary to represent the largest encoded integer in the stream.
+
+The `bit_width` MUST be sufficient to represent the largest encoded integer on the
+stream or the result is undefined.
+
+The first 4 bytes of the stream MUST represent a little-endian unsigned integer (`uint32`)
+denoting the length of the rest of the stream. For example, `[4u8, 0, 0, 0]` announces
+that the stream has a total of `4 + 4 = 8` bytes. The first 4 bytes are only used for this purpose.
```

Comment:
Why `4 + 4`? Do you mean including the first 4 bytes?

Also, the sentence "The first 4 bytes are only used for this purpose" isn't very informative.

Reply:
yes, the first 4 + the announced 4. One idea is use a different number (i.e. not `4`), so that it is more clear where these come from.

 I will remove the other sentence.